File: simulation/tests/test_time.py
import argparse
import os
import time

from simulation import *

DEFAULT_INPUT_FOLDER = 'input_data'
DEFAULT_OUTPUT_FOLDER = 'output'
parser = argparse.ArgumentParser()
parser.add_argument('--input', default=os.path.join(DEFAULT_INPUT_FOLDER, 'd.txt'), help='Input file.')
parser.add_argument('--plan', default=os.path.join(DEFAULT_OUTPUT_FOLDER, 'd.txt'), help='Plan file.')
args = parser.parse_args()

if __name__ == '__main__':
    start = time.time()
    sim = Simulation(args.input)
    print(f'Simulation constructor: {time.time() - start:.4f}s')

    start = time.time()
    instance = sim.create_instance()
    print(f'create_instance: {time.time() - start:.4f}s')

    # start = time.time()
    # instance.read_plan(args.plan)
    # print(f'read_plan: {time.time() - start:.4f}s')

    start = time.time()
    instance.create_plan_default()
    print(f'create_plan_default: {time.time() - start:.4f}s')

    start = time.time()
    instance.run()
    print(f'run: {time.time() - start:.4f}s')

File: simulation/tests/test_time.cpp
#include <chrono>
#include <iostream>
#include <string>
#include <vector>

#include "simulation.hpp"

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    auto &&input_file = args[0];
    auto &&plan_file = args[1];

    auto start = std::chrono::high_resolution_clock::now();
    Simulation simulation{input_file};
    auto end = std::chrono::high_resolution_clock::now();
    std::cout << "Simulation constructor: " << std::chrono::duration<double>{end - start}.count() << "s\n";

    start = std::chrono::high_resolution_clock::now();
    auto &&simulation_instance = simulation.create_instance();
    end = std::chrono::high_resolution_clock::now();
    std::cout << "create_instance: " << std::chrono::duration<double>{end - start}.count() << "s\n";

    start = std::chrono::high_resolution_clock::now();
    simulation_instance.create_plan_default();
    end = std::chrono::high_resolution_clock::now();
    std::cout << "create_plan_default: " << std::chrono::duration<double>{end - start}.count() << "s\n";

    //start = std::chrono::high_resolution_clock::now();
    //simulation_instance.read_plan(plan_file);
    //end = std::chrono::high_resolution_clock::now();
    //std::cout << "read_plan: " << std::chrono::duration<double>{end - start}.count() << "s\n";

    start = std::chrono::high_resolution_clock::now();
    simulation_instance.run();
    end = std::chrono::high_resolution_clock::now();
    std::cout << "run: " << std::chrono::duration<double>{end - start}.count() << "s\n";
}
